src/tests/test_experiments.py

```diff
@@ -135,8 +135,31 @@ def test_simple_experiment(experiment_class, matrix_storage_class):
                 ]
             )
             assert num_evaluations > 0
+        
+        # 5. subset evaluations linked to subsets and predictions linked to
+        #    models, for training and testing
+        for set_type in ("train", "test"):
+            num_evaluations = len(
+                [
+                    row
+                    for row in db_engine.execute(
+                        """
+                        select * from {}_results.evaluations e
+                        join model_metadata.models using (model_id)
+                        join model_metadata.subsets using (subset_hash)
+                        join {}_results.predictions p on (
+                            e.model_id = p.model_id and
+                            e.evaluation_start_time <= p.as_of_date and
+                            e.evaluation_end_time >= p.as_of_date)
+                        """.format(
+                            set_type, set_type
+                        )
+                    )
+                ]
+            )
+            assert num_evaluations > 0
```

Comment:
predictions:evaluations is many:one, so this will return multiple rows, unless we group by model_id, which we can do.

Reply:
group by model_id & subset_hash, that is

Reply:
OK. I may have misunderstood or now be misremembering :man_shrugging: 

My point was just that it's unnecessary to build a Python list of rows that you just want to count.

For example, even without adjusting your query, I believe you can use the `ResultProxy` method `rowcount` [[docs]](https://kite.com/python/docs/sqlalchemy.engine.result.ResultProxy.rowcount).

But this is from my armchair :smile: 

…And this is not the most important thing to worry about!

Reply:
…But sorry not `rowcount`. Anyway, you know! It can be in the query. Or as a slight improvement your Python can just count, rather than build a list.

But now that I'm on record … really, feel free to carry on :+1: